shared/js/intl_helper.js

```diff
@@ -0,0 +1,148 @@
+(function(global) {
+  'use strict';
+
+  /*************************
+
+  IntlHelper.defineObject('shortDate', 'datetime', {
+    year: 'numeric',
+    month: 'long',
+  });
+
+  function setValue() {
+    var formatter = IntlHelper.getObject('shortDate');
+    console.log(formatter.format(new Date()));
+  }
+
+  setValue();
+
+  IntlHelper.on('shortDate', setValue);
+  
+   ****************************/
+  var objectCache;
+
+  var knownObjects = [
+    'datetime',
+    'number',
+  ];
+
+  global.IntlHelper = {
+    defineObject: function(name, type, options) {
+      if (knownObjects.indexOf(type) === -1) {
+        throw new Error('Unknown type: ' + type);
+      }
+
+      if (!objectCache) {
+        objectCache = new Map();
+      }
+
+      if (objectCache.has(name)) {
+        throw new Error(
+          'Intl Object with name "' + name + '" already exists');
+      }
+
+      objectCache.set(name, {
+        options,
+        type,
+        object: undefined,
+        listeners: new Set()
+      });
+    },
+
+    getObject: function(name) {
+      if (!objectCache.has(name)) {
+        throw new Error('Intl Object with name "' + name + '" is not defined');
+      }
+      const obj = objectCache.get(name);
+
+      if (obj.object === undefined) {
+        switch (obj.type) {
+          case 'number':
+            obj.object = Intl.NumberFormat(
+              navigator.languages,
+              obj.options
+            );
+            break;
+          case 'datetime':
+            const options = Object.assign({}, obj.options);
+            if (options.hour) {
+              options.hour12 = navigator.mozHour12;
+            }
+            console.log('new Format');
+            console.log(options);
+            obj.object = Intl.DateTimeFormat(
+              navigator.languages,
+              options
+            );
+            break;
+          default:
+            throw new Error('Unknown type: ' + obj.type);
+        }
+      }
+      return obj.object;
+    },
+    on: function(names, cb) {
+      if (typeof names === 'string') {
+        names = [names];
+      }
+      for (var name of names) {
+        if (!objectCache.has(name)) {
+          throw new Error(
+            'Intl Object with name "' + name + '" is not defined');
+        }
+
+        var obj = objectCache.get(name);
+
+        obj.listeners.add(cb);
+      }
+    },
+    off: function(name, cb) {
+      var obj = objectCache.get(name);
+
+      obj.listeners.remove(cb);
+    },
+    emit: function(type, ...args) {
+      var affectedCallbacks = new Set();
+
+      objectCache.forEach((obj, name) => {
+        if (obj.type === type) {
+          obj.listeners.forEach(cb => affectedCallbacks.add(cb));
+        }
+      });
+
+      affectedCallbacks.forEach(cb => {
+        cb(...args);
+      });
```

Comment:
Nit: you don't need curly braces if there's only one expression.

    affectedCallbacks.forEach(cb => cb(...args));

However, should we somehow handle callback which throw?